src/Common/tests/Tests/System/Net/VirtualNetworkStreamTest.cs

```diff
@@ -109,10 +119,19 @@ public void VirtualNetworkStream_MultiThreadIntegrityTest_Ok()
             {
                 Parallel.For(0, 100000, async (int i) =>
```

Comment:
This is problematic and isn't doing what you're hoping it is.  The delegate is an Action<int>, which means this async method is void-returning.  The moment the body awaits on something that's not yet completed, the delegate will return, and as far as the Parallel.For is concerned, that delegate finished its work.  See http://blogs.msdn.com/b/pfxteam/archive/2012/02/08/10265476.aspx for more details.  You'll need to either change the awaits in the delegate to be blocking waits, or use something other than Parallel.For.